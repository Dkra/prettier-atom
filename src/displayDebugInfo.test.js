// @flow
const displayDebugInfo = require('./displayDebugInfo');

test('it displays a notification on Atom with package information', () => {
  let title;
  let options;

  atom = {
    getVersion: jest.fn(),
    config: {
      get: jest.fn(),
    },
    notifications: {
      addInfo: jest.fn((msg, opts) => {
        title = msg;
        options = opts;
      }),
    },
  };

  displayDebugInfo();

  expect(atom.notifications.addInfo).toHaveBeenCalled();
  expect(title).toContain('prettier-atom');
  // $FlowFixMe
  expect(options.detail).toMatchSnapshot();
  // $FlowFixMe
  expect(options.dismissable).toBe(true);
});
