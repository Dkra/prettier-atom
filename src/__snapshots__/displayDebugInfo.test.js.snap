// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it displays a notification on Atom with package information 1`] = `
"Atom version: undefined
prettier-atom version: 0.30.0
prettier version: 1.3.1
prettier-eslint version: 6.2.0
prettier-atom configuration: undefined
"
`;
